### PR TITLE
OpenRTB specification, communication boilerplate code, gRPC.

### PR DESCRIPTION
**Pull request by arkady-emelyanov**

### CODE REVIEW

File: internal/api/openrtb.proto [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    optional Banner banner = 2;
+
+    optional double bidfloor = 3;
+}
+
+message Banner {
+    //
+    optional string id = 1;
+
+    //
```
Comment (spaghettifunk) [inline internal/api/openrtb.proto]: should we add the comments to specify what are the meaning of those variable ? except for the most obvious ones like: zip, city, etc

File: internal/frontend/bid.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+			w.WriteHeader(http.StatusNoContent)
+			w.Write(nil)
+			return
+		}
+
+		// if bidResponse is not received from bidder-backend
+		// respond with no-bid status.
+		if resBid.Response == nil {
+			log.Printf("Backend decided not to bid")
```
Comment (spaghettifunk) [inline internal/frontend/bid.go]: considering that the latency in the bidder needs to be very low, I suggest to use the package [zero-log](https://github.com/rs/zerolog) and channel the logs out there.

File: pkg/middleware/logging.go [hunk 3]
```diff
@@ -6,16 +6,16 @@ import (
...
 
 // WithLogging allows to log messages around the request made
 func WithLogging(next http.HandlerFunc) http.Handler {
 	fn := func(w http.ResponseWriter, r *http.Request) {
-		log.Println(startLogMsg + r.RemoteAddr)
-		// Defer is slow, should come up with something smarter
-		defer log.Println(finishLogMsg + r.RemoteAddr)
+		log.Printf(startLogMsg, r.RemoteAddr)
+		// FIXME: Defer is slow, should come up with something smarter
+		defer log.Printf(finishLogMsg, r.RemoteAddr)
```
Comment (spaghettifunk) [inline pkg/middleware/logging.go]: see my comment above for logs

File: internal/frontend/bid.go (hunk 2 above)
Comment (arkady-emelyanov) [reply internal/frontend/bid.go]: Yes, will update tomorrow. Really good addition.

File: internal/api/openrtb.proto (hunk 1 above)
Comment (arkady-emelyanov) [reply internal/api/openrtb.proto]: Yes, already started it. Will update eventually :)

**Pull request merged**